### ISSUE TITLE
Combining knots and NoInterp

### ISSUE THREAD

**Issue by matthieugomez** [tlycken/Interpolations.jl#60]
I'm trying to use `interpolate` with knots & `NoInterp` along some dimension. 

- With BSpline, using knots as a first argument returns an error
 
 ```julia
 A = [0.0593468  0.73381 ; 0.298259   0.135074]
 itp = interpolate((1:2, 2:3), A, BSpline{Linear}, OnGrid)
 ```
I'm not sure whether `knots` are available for `Spline` 
- With Grid, combining `knots` and `Tuple{Gridded{Linear}, Gridded{NoInterp}` returns unexpected results

 ```julia
 A = [0.0593468  0.73381 ; 0.298259   0.135074]
 itp = interpolate((1:2, 2:3), A, Tuple{Gridded{Linear}, Gridded{NoInterp}})
 # itp[1, 2] returns A[1, 2] but should return A[1, 1]
 itp[1, 2]
 # 0.7338095900816122
  ```

**Comment by timholy** [comment]
`knots` are only available with Gridded. What's called `BSpline` is really "uniform B-spline."

**Comment by tlycken** [comment]
For this use case, though, it looks like you could use the scaling approach that will be added by #47.

**Comment by matthieugomez** [comment]
Yes I only need the scaling, so that would work. The Gridded example is a bug though, right?

**Comment by timholy** [comment]
> The Gridded example is a bug though, right?

No, `NoInterp` means "just pass the index through." You might be looking for `Gridded{Constant}`.

**Comment by timholy** [comment]
Might need some API fixing though---we should probably throw an error for those knots. Care to submit a pull request?

**Comment by matthieugomez** [comment]
I've looked into the (terse!) code for `Grid` but I'm not sure how to modify it. 
I think there should be an error when, along some dimension `d`, the type is `Gridded{NoInterp}` but the knots is not `1:size(A, d)`.

**Comment by timholy** [comment]
Yep, that's how I'd do it. Nice digging!

**Comment by timholy** [comment]
Fixed by #61.

**Comment by matthieugomez** [comment]
Thanks!
